authenticator.go

```diff
@@ -0,0 +1,87 @@
+package main
+
+import (
+	"bytes"
+	"encoding/hex"
+	"io/ioutil"
+	"net/http"
+
+	p2pcrypto "github.com/libp2p/go-libp2p-crypto"
+
+	log "github.com/sirupsen/logrus"
+)
+
+type Authenticator interface {
+	AuthenticateRequest(r *http.Request) bool
+}
+
+type P2PAuthenticator struct {
+	// B64 encoded public keys of identities whose logs are allowed to be
+	// ingested
+	allowedPubKeys []string
```

Comment:
It seems we never reference this property. Let's remove it since we can always iterate over all pubKey strings using the `pubKeyMap` like so:

```
for allowedPubKey := range pubKeyMap {
  ...
}
```

Reply:
removed